src/platform-aws.cpp

```diff
@@ -1082,3 +1083,18 @@ void PlatformAWS::sort_rails(struct fi_info **info_list, size_t num_rails, size_
 
 	return;
 }
+
+bool PlatformAWS::is_ec2_instance() {
+	// Validate that platform parameter wasn't set inconsistently between libraries
+	if (ofi_nccl_platform.get_source() == ParamSource::API) {
+		NCCL_OFI_WARN("Platform parameter was set via API call, which may indicate "
+		              "inconsistent configuration between network and tuner libraries");
+	}
+
+	PLATFORM platform_override = ofi_nccl_platform.get();
+	if (platform_override != PLATFORM::AUTO) {
+		return platform_override == PLATFORM::AWS;
```

Comment:
Why should this be in each platform file?  What's the point of the platform manager code if you're just going to have a bunch of the selection logic elsewhere?